### ISSUE TITLE
Bug: string id is always required convert to int64

### ISSUE THREAD

**Issue by fitzix** [facebookincubator/ent#384]
I confirmed that the problem of string id in migration #372 has been resolved, but when running the query `All` scan, the string edge field is still required to be converted to int64

>  sql: Scan error on column index 2, name "user_card": converting driver.Value type string ("WKE81uvM88lJD_9W") to a int64: invalid syntax

```go
// Fields of the User.
func (User) Fields() []ent.Field {
	return []ent.Field{
		field.String("id").MaxLen(16),
	}
}

// Edges of the user.
func (User) Edges() []ent.Edge {
    return []ent.Edge{
        edge.To("card", Card.Type).Unique(),
    }
}

// Fields of the User.
func (Card) Fields() []ent.Field {
	return nil
}

// Edges of the Card.
func (Card) Edges() []ent.Edge {
    return []ent.Edge{
        edge.From("owner", User.Type).
            Ref("card").
            Unique().
            // We add the "Required" method to the builder
            // to make this edge required on entity creation.
            // i.e. Card cannot be created without its owner.
            Required(),
    }
}

client.User.Query().WithCard().All(context.Background())
```

**Comment by hantmac** [comment]
`id` used as `PrimaryKey` and must be `int64`, if your wanna to use string as unique, can use `uuid` instead.

**Comment by fitzix** [comment]
> `id` used as `PrimaryKey` and must be `int64`, if your wanna to use string as unique, can use `uuid` instead.

> Also, please note that in this case, if you use `field.String` as a PK, you must limit its size (with `MaxLen`).
> _Originally posted by @a8m in https://github.com/facebookincubator/ent/issues/372#issuecomment-594408479_

This is not right, as @am8 said, string primary keys are allowed, there is no requirement that id must be `int64` or `uuid`

The documentation example also has this usage https://entgo.io/docs/schema-fields/#id-field

**Comment by hantmac** [comment]
> 哈哈 又是你😅
> 
> > `id` used as `PrimaryKey` and must be `int64`, if your wanna to use string as unique, can use `uuid` instead.
> 
> > Also, please note that in this case, if you use `field.String` as a PK, you must limit its size (with `MaxLen`).
> > _Originally posted by @a8m in [#372 (comment)](https://github.com/facebookincubator/ent/issues/372#issuecomment-594408479)_
> 
> This is not right, as @am8 said, string primary keys are allowed, there is no requirement that id must be `int64` or `uuid`
> 
> The documentation example also has this usage https://entgo.io/docs/schema-fields/#id-field

HAHA, I do rarely use strings as primary keys in development. And I tested mysql passed using `id` as string.
![image](https://user-images.githubusercontent.com/7600925/76404820-1bd57280-63c2-11ea-9f14-1ea6fdb32eb3.png)

**Comment by fitzix** [comment]
Yes, the string id is really not recommended as the primary key, but it is allowed in ent and sometimes it is needed. The problem is that the string id will make the associated field also a string, but when scanning into the struct, it still was asked to convert to int64

**Comment by hantmac** [comment]
I created some data
```
func CreatePet(ctx context.Context, client *ent.Client) (*ent.Pet, error) {
	p := client.Pet.Create().SetName("doc2").SetID("no2").SaveX(ctx)

	return p, nil
}
```
and did a test to scan `pet`  into the struct

```
	res, _ := client.Pet.Query().All(ctx)
	fmt.Println(res) //The res is `Pet(id=no1, name=doc) Pet(id=no2, name=doc2)]`
```
And the schema of pet is:
```
+-------+--------+--------+----------+----------+---------+---------------+-----------+-----------------------+------------+
        | Field |  Type  | Unique | Optional | Nillable | Default | UpdateDefault | Immutable |       StructTag       | Validators |
        +-------+--------+--------+----------+----------+---------+---------------+-----------+-----------------------+------------+
        | id    | string | true   | false    | false    | false   | false         | true      | json:"id,omitempty"   |          2 |
        | name  | string | false  | false    | false    | false   | false         | false     | json:"name,omitempty" |          0 |
        +-------+--------+--------+----------+----------+---------+---------------+-----------+-----------------------+------------+
```

**Comment by fitzix** [comment]
The problem appeared in Eager Loading, such as O2O edge
execute `client.User.Query().WithPet().All(ctx)`

**Comment by a8m** [comment]
Hey, it's probably a bug. 
The reason for the conversion (`string` to `int`) is mentioned [here](https://github.com/facebookincubator/ent/issues/127#issuecomment-583834177). It should be fixed soon - see my comment in the link. 

I'm not pushing a change now, because we have a large PR (#377) that is supposed to land in the upcoming days and I want to avoid template conflicts. 

I'll keep this issue open and update it once it's fixed.

**Comment by hantmac** [comment]
@fitzix  got  it.

**Comment by a8m** [comment]
Closed with #389

Thanks!
